modules/memberships/class-jetpack-memberships.php

```diff
@@ -270,10 +307,10 @@ public function render_button( $attrs ) {
 					array( 'wp-block-button' )
 				)
 			),
-			esc_url( $button_url ),
+			esc_url( $this->get_subscription_url( $plan_id ) ),
 			isset( $attrs['submitButtonClasses'] ) ? esc_attr( $attrs['submitButtonClasses'] ) : 'wp-block-button__link',
 			esc_attr( $button_styles ),
-			wp_kses( $data['button_label'], self::$tags_allowed_in_the_button ),
+			wp_kses( $button_label, self::$tags_allowed_in_the_button ),
 			isset( $attrs['submitButtonAttributes'] ) ? sanitize_text_field( $attrs['submitButtonAttributes'] ) : '' // Needed for arbitrary target=_blank on WPCOM VIP.
```

Comment:
I'm not clear on where and how the `submitButtonAttributes` for WPCOM VIP sites are set. Can anyone shed any light on this and see if these changes might introduce other VIP related issues?

Reply:
Turns out these come from direct calls to `Jetpack_Memberships::get_instance()->render_button()`. As they won't supply inner block content, they'll be treated as deprecated buttons by `render_button`. 

I've removed the unnecessary handling of `submitButtonAttributes` while rendering the latest version of the block.